### DISCUSSION TITLE
Cant get page script working...

### DISCUSSION THREAD

**Discussion by ghost** [requarks/wiki#3188] [Help / Questions] [answered]
Am I missing something on getting page scripts running? I have a Markdown page with an HTML text field: < span id="foo" >bar< /span >.

In the page script, I'm attempting to make a change to the text field with a JS script:

< script defer >
document.getElementById("foo").textContent = "foo bar";
</ script >

When I save the page and close, the text still only shows "bar". Any idea why this isn't working?

Edit: dev console shows Uncaught TypeError: Cannot set property "textContent" of null, but inspect element shows that the span is IDed as expected.

**Comment by NGPixel** [comment, marked as answer]
That's because you're executing code before the page is initialized. You need to register your code as a hook:
```js
window.boot.register('vue', () => {
  // your code here
})
```

**Reply by ghost** [reply]
Thanks so much for this. I'm assuming I can install Vue globally on my VPS, then import it in the HTML head injection inside a < script > ? Sorry for the silly questions, relatively new to Vue and wiki.js :)

**Reply by NGPixel** [reply]
No, Vue is a client library, not something you install on your server. There's nothing to import, it's already there. You just put your code in the example I provided.

**Reply by srob411** [reply]
Can’t get that to work with my wiki.js (V 2.5.170 on Debian/Docker). Tried exactly as described above:

-	Put \<span id="foo"\>bar\</span\> in the Markdown of the page 
-	Wrote this into the script section of the page:
```
<script>
window.boot.register('vue', () => {
  // alert("Hello world");
  document.getElementById("foo").textContent = "foo bar";
})
</script>
```

If I uncomment the alert, that is working. But I cannot access the foo element, it is undefined: 

```
Cannot set property 'textContent' of null in document.getElementById("foo").textContent = "foo bar";
```

What am I missing?

**Reply by Heziode** [reply]
Same issue has @srob411

**Reply by NGPixel** [reply]
Try adding a delay:

```js
window.boot.register('vue', () => {
  setTimeout(() => {
    document.getElementById('foo').textContent = 'foo bar'
  }, 1000)
})
```

**Reply by Heziode** [reply]
This is what I just did, it works, but it is disgusting
